airbyte-integrations/connectors/source-harvest/source_harvest/streams.py

```diff
@@ -0,0 +1,361 @@
+        return f"estimates/{stream_slice['parent_id']}/messages"
+
+
+class EstimateItemCategories(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "estimate_item_categories"
+
+    def path(self, **kwargs) -> str:
+        return "estimate_item_categories"
+
+
+class Expenses(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "expenses"
+
+    def path(self, **kwargs) -> str:
+        return "expenses"
+
+
+class ExpenseCategories(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "expense_categories"
+
+    def path(self, **kwargs) -> str:
+        return "expense_categories"
+
+
+class Tasks(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "tasks"
+
+    def path(self, **kwargs) -> str:
+        return "tasks"
+
+
+class TimeEntries(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "time_entries"
+
+    def path(self, **kwargs) -> str:
+        return "time_entries"
+
+
+class UserAssignments(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "user_assignments"
+
+    def path(self, **kwargs) -> str:
+        return "user_assignments"
+
+
+class TaskAssignments(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "task_assignments"
+
+    def path(self, **kwargs) -> str:
+        return "task_assignments"
+
+
+class Projects(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "projects"
+
+    def path(self, **kwargs) -> str:
+        return "projects"
+
+
+class Roles(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "roles"
+
+    def path(self, **kwargs) -> str:
+        return "roles"
+
+
+class Users(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    data_field = "users"
+
+    def path(self, **kwargs) -> str:
+        return "users"
+
+
+class BillableRates(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    data_field = "billable_rates"
+    parent_stream = Users
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/billable_rates"
+
+
+class CostRates(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    data_field = "cost_rates"
+    parent_stream = Users
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/cost_rates"
+
+
+class ProjectAssignments(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    data_field = "project_assignments"
+    parent_stream = Users
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/project_assignments"
+
+
+class ReportsBase(HarvestStreamWithPagination, ABC):
```

Comment:
If we can use `from` and `to` params, than why these streams can be only Full Refresh Sync?

Reply:
Well, we could not get any cursor data from the response. But it's indeed possible to generate a cursor value on our side only. I'll do it.

Reply:
I found that it's not possible (or at least not really convenient) to do with CDK.
Also acceptance tests won't pass even if this flow is implemented.